### PR TITLE
FrontEnd Create Function to check users is auth and Store in Redux

### PR DESCRIPTION
**Pull request by AhmedQeshta**
### Create Function to check  user/gym isAuth

- use `jwt-decode`
- store the info in store redux to know if is auth not
- store `role, id, name, isLoggedIn‏` in redux

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -14,8 +18,22 @@ import {
   NotfoundPage,
   HomePage,
 } from "./Pages";
+import { setAuth, setLogout } from "./Store/Slices/checkAuthSlice";
```
Comment (LinaYahya) [inline client/src/App.js:21]:
```suggestion
import { setAuth, setLogout } from "./Store/Slices/";
```

File: client/src/App.js [hunk 2]
```diff
@@ -14,8 +18,22 @@ import {
...
 
 function App() {
+  const [cookies] = useCookies();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const checkToken = () => {
+      const { token } = cookies;
+      if (!token) dispatch(setLogout());
```
Comment (LinaYahya) [inline client/src/App.js:30]: Why you do this?!
As ther'e is no token So the state will keep the initial state,
the only place to `dispatch(setLogout())` at logout button

File: client/src/index.js [hunk 3]
```diff
@@ -2,17 +2,20 @@ import React from "react";
...
 
 const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
   <Theme>
-    <Router>
-      <Provider store={store}>
-        <App />
-      </Provider>
-    </Router>
+    <CookiesProvider>
```
Comment (LinaYahya) [inline client/src/index.js:13]: why are you using this module,
Can you get cookies from js as `document.cookies`?
if not check this module I think it's better than the one you used
[js-cookie](https://www.npmjs.com/package/js-cookie)

File: client/src/App.js [hunk 4]
```diff
@@ -14,8 +18,22 @@ import {
...
+      const { role, gymName, gymID } = jwt_decode(token);
```
Comment (LinaYahya) [inline client/src/App.js:31]: what if it's a userCookie?!

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/index.js (hunk 3 above)
Comment (AhmedQeshta) [reply client/src/index.js:13]: I want to use [js-cookie](https://www.npmjs.com/package/js-cookie)

**Review by rawaneyad — APPROVED** [review]
Great Job Ahmed👍🏼

File: client/src/App.js [hunk 5]
```diff
@@ -15,7 +19,34 @@ import {
...
+        dispatch(
+          setAuth({
+            id,
+            name,
+            role,
+            isLoggedIn: true,
+          })
+        );
+      } catch (error) {
+        console.log(error); // ?? handle error
```
Comment (LinaYahya) [inline client/src/App.js:44]:
```suggestion
         dispatch(
          setAuth({
            id: null,
            name: null,
            role: null,
            isLoggedIn: false,
          })
```

**Review by LinaYahya — APPROVED** [review]
Great Job Ahmed

File: client/src/App.js (hunk 5 above)
Comment (AhmedQeshta) [reply client/src/App.js:44]: ok I will do it now

Comment (AhmedQeshta) [reply client/src/App.js:44]: Can use function logout

**Review by AFahgan — APPROVED** [review]
Great job 👍

**Pull request merged**